custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js

```diff
@@ -337,17 +345,123 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
         }
     };
 
+    // mobile helpers
+    vm.showSortPopup = false;
+    vm.tooltipDisplayed = -1; // '-1' when none of the tooltips are displayed in sort popup
+    vm.requestDataStartingFrom = 0; // could be any multiple of 10
+    vm.dataSortingDirection = 0; // '0' -> asc, '1' -> desc
+    vm.sortingColumn = 0;
+    vm.sortableInputKpiData = [];
+    vm.showSortingInfo = function (tooltipNumber) {
+        // when clicked on 'i' icon, get the column number on which info is requested and show only corresponding info.
+        vm.tooltipDisplayed = tooltipNumber;
+    };
+    vm.toggleSortPopup = function (event) {
+        vm.tooltipDisplayed = -1;
+        vm.showSortPopup = !vm.showSortPopup;
+        event.stopPropagation();
```

Comment:
out of curiosity why is this needed?

Reply:
At the top level element I have added a click event, which when triggered closes sort popup. But we dont need the click event which triggers the popup opening to bubble up to the top, which will close it. So, this prevents that click event from bubbling up the dom

Reply:
got it. that feels comment-worthy to me